<launch>
  <include file="$(find do_description)/launch/mobile_robot_launch.launch"/>
  
 <!-- joy node -->
  <node respawn="true" pkg="joy"
        type="joy_node" name="turtle_joy" >
    <param name="dev" type="string" value="/dev/input/js0" />
    <param name="deadzone" value="0.12" />
  </node>

 <!-- Axes -->
  <param name="axis_linear" value="1" type="int"/>
  <param name="axis_angular" value="0" type="int"/>
  <param name="axis_vertical" value="4" type="int"/>
  <param name="axis_horizontal" value="3" type="int"/>
  <param name="scale_linear" value="2" type="double"/>
  <param name="scale_angular" value="2" type="double"/>
  <param name="scale_vertical" value="1" type="double"/>
  <param name="scale_horizontal" value="1" type="double"/>
  <node pkg="do_operation" type="teleop_robot" name="teleop"/>
  <node pkg="do_operation" type="teleop_robot2" name="teleop2"/>

  <rosparam file="$(find do_operation)/config/robot_control.yaml" command="load"/>

  <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" ns="/mobile_robot" args="joint1_position_controller joint2_position_controller joint_state_controller"/>

  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
    respawn="false" output="screen">
    <remap from="/joint_states" to="/mobile_robot/joint_states" />
  </node>

</launch>